File: frontend/src/constants/GroundTaskStatus.ts
export const GROUND_TASK_STATUSES = ["PLANNED", "SIGNED", "FINISHED", "BLOCKED"] as const;
export type GroundTaskStatusValue = (typeof GROUND_TASK_STATUSES)[number];

export const GROUND_TASK_STATUS_TEXT: Record<GroundTaskStatusValue, string> = {
  PLANNED: "待签收",
  SIGNED: "已签收",
  FINISHED: "已完成",
  BLOCKED: "阻塞"
};

// 已签收/已完成任务在延误重排时保持原计划时点。
export const ACKNOWLEDGED_TASK_STATUSES: GroundTaskStatusValue[] = ["SIGNED", "FINISHED"];

File: backend/src/constants/GroundTaskStatus.go
package constants

// GroundTaskStatus tracks dispatch / sign-off lifecycle.
type GroundTaskStatus string

const (
	TaskPlanned   GroundTaskStatus = "PLANNED"   // 已派发未签收
	TaskSigned    GroundTaskStatus = "SIGNED"    // 班组已签收
	TaskFinished  GroundTaskStatus = "FINISHED"  // 已完成
	TaskBlocked   GroundTaskStatus = "BLOCKED"   // 阻塞
)

var GroundTaskStatuses = []GroundTaskStatus{
	TaskPlanned, TaskSigned, TaskFinished, TaskBlocked,
}

var GroundTaskStatusText = map[GroundTaskStatus]string{
	TaskPlanned:  "待签收",
	TaskSigned:   "已签收",
	TaskFinished: "已完成",
	TaskBlocked:  "阻塞",
}

// Acknowledged means the crew already signed for the task; delay replanning
// must keep these time points untouched.
func (s GroundTaskStatus) Acknowledged() bool { return s == TaskSigned || s == TaskFinished }
func (s GroundTaskStatus) Finished() bool     { return s == TaskFinished }

func (s GroundTaskStatus) Text() string { return GroundTaskStatusText[s] }
